### PR TITLE
Enable additional linters: bodyclose, noctx, dup, forbidigo,godot,gomnd

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR enables new linters: ~bodyclose~, noctx, dup, forbidigo,godot,gomnd.

bodyclose has some false positives, and I decided to not include it.

## How can this be tested?

unit-tests, CI

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by 0sewa0 — CHANGES_REQUESTED** [review]
commented about a few minor issues, but in general

I do not like `godot`, or at least not enabled for the whole codebase, enabling it for the CRDs make sense, but it still does not solve my main problem:
- It does not guarantee a nice comment, in the few examples I highlighted it actually makes them worse, if we only fix this linter.
  - If there more linters that enforce "nice doc strings" then we should enable them all at once, and fix the doc strings properly

File: pkg/clients/dynatrace/dynatrace_client.go [hunk 1]
```diff
@@ -56,7 +57,8 @@ const (
 // makeRequest does an HTTP request by formatting the URL from the given arguments and returns the response.
 // The response body must be closed by the caller when no longer used.
 func (dtc *dynatraceClient) makeRequest(url string, tokenType tokenType) (*http.Response, error) {
-	req, err := http.NewRequest(http.MethodGet, url, nil)
+	// TODO: introduce ctx into dynatrace client
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply pkg/clients/dynatrace/dynatrace_client.go]: fixed in 31ffdd24

File: pkg/clients/dynatrace/client.go [hunk 2]
```diff
@@ -100,19 +100,19 @@ const (
...
-	// OsSolaris = "solaris"
+	// OsSolaris = "solaris".
 )
 
 // Known installer types.
 const (
 	InstallerTypeDefault = "default"
 	// Commented for linter, left for further reference
-	// InstallerTypeUnattended = "default-unattended"
+	// InstallerTypeUnattended = "default-unattended".
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply pkg/clients/dynatrace/client.go]: removed.

File: pkg/clients/dynatrace/dynatrace_client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply pkg/clients/dynatrace/dynatrace_client.go]: fixed

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**